nginx.conf

```diff
@@ -129,6 +129,38 @@ http {
         error_page 301 302 307 = @handle_redirect;
       }
 
+      location ~ /html/(?<proxied_uri>.*) {
+        # TODO: make this internal when it gets called from proxy_based_on_content_type.
+        set $upstream $proxied_uri$is_args$args;
+
+        set_by_lua_file $hypothesis_header ./lua/h_header.lua; 
+
+        # Insert embeded Hypothesis client into header and add base tag so remaining
+        # relative links are relative to the third party url.
+        subs_filter '(<head>|<head [^>]*>)' '<base href="$proxied_uri">$1$hypothesis_header' r;
+
+        sub_filter_last_modified on;
+
+        proxy_set_header Accept-Encoding ""; 
+        proxy_ssl_server_name on;
```

Comment:
From here on down you can ignore. Pretty much every endpoint does this. Handle redirects, don't send cookies to third party server, don't let third party server set cookies, etc.